homeassistant/components/hyperion/strings.json

```diff
@@ -50,5 +50,33 @@
         }
       }
     }
+  },
+  "entity": {
+    "switch": {
+      "all": {
+        "name": "Component all"
+      },
+      "smoothing": {
+        "name": "Component smoothing"
+      },
+      "blackbar_detection": {
+        "name": "Component blackbar detection"
+      },
+      "forwarder": {
+        "name": "Component forwarder"
+      },
+      "boblight_server": {
+        "name": "Component boblight server"
+      },
+      "platform_capture": {
+        "name": "Component platform capture"
+      },
+      "led_device": {
+        "name": "Component LED device"
+      },
+      "usb_capture": {
+        "name": "Component USB capture"
```

Comment:
The existing names are:

```
    KEY_COMPONENTID_ALL: "All",
    KEY_COMPONENTID_SMOOTHING: "Smoothing",
    KEY_COMPONENTID_BLACKBORDER: "Blackbar Detection",
    KEY_COMPONENTID_FORWARDER: "Forwarder",
    KEY_COMPONENTID_BOBLIGHTSERVER: "Boblight Server",
    KEY_COMPONENTID_GRABBER: "Platform Capture",
    KEY_COMPONENTID_LEDDEVICE: "LED Device",
    KEY_COMPONENTID_V4L: "USB Capture",
```

Is the extra "Component" worth it vs sticking with the existing shorter names?

Reply:
As far as I have seen, the current name was composed by instance_name + "Component" + that component name from that list so that's what this will result into. But I'm happy to change if you have a better idea

Reply:
I checked, you're right! Favor keeping the names consistent unless we have reason to change them, and I think since these are human-facing little-endian in English is probably slightly better! i.e. How about we use "USB Capture Component" (etc) in the translations, and it'll then sound slightly better and exactly match the prior names?

Reply:
I don't really mind about names, if you agree with doing "... component" on everything I can change that tomorrow or Monday. If you have other names (I mean "all component" doesn't sound nice) please create suggestions

Reply:
> If you have other names (I mean "all component" doesn't sound nice) please create suggestions

I was simply saying lets try to keep the names identical unless we have reason to change (incl. capitalization). I agree "All Component" is not nice, but I couldn't think of anything better then, and also can't think of something better now!

Just checked on my running instance, the names are like: `Sitting Room Hyperion Component USB Capture`

```py
def _component_to_switch_name(component: str, instance_name: str) -> str:
    """Convert a component to a switch name."""
    return (
        f"{instance_name} "
        f"{NAME_SUFFIX_HYPERION_COMPONENT_SWITCH} "
        f"{KEY_COMPONENTID_TO_NAME.get(component, component.capitalize())}"
    )
```

This highlights an issue with this PR -- I think we end up losing the instance name. So if a Hyperion configuration has more than a single LED output instance, the switches would end up with the same name (i.e. we could have 3 `Component USB capture` switches without an obvious way to tell them apart).

Perhaps there's a way to keep the translations (which is a big improvement!), but prefix something in front of them ...

Reply:
The instance name is now replaced with the device name. So now the entity is created like "<device name> <translated entity name>". The only way to create dynamic stuff is via customizing the device name.

Reply:
> The instance name is now replaced with the device name. 

OK, that's probably fine then since the device name is the Hyperion instance name, so I think it should be possible to get the names to exactly match before vs after this PR without too much trouble.

Reply:
Oh and since everything has unique id, it doesn't matter what we rename it to, so if you still have improvements, please let me know

Reply:
> Oh and since everything has unique id, it doesn't matter what we rename it to

Yup, only matters for added devices. I'm happy here once the names exactly match what they were. I think the only delta left is probably the capitalization -- but just verify on your end that they show up the same before/after, and I'll approve!

Reply:
Hehe, fun fact, I don't have Hyperion, I'm just on a quest to get every integration with entity translations. Theoretically this should work. I think Hyperion also has tests so if something major changed we would've seen it.

The capitalization should be only the first letter capitalized (except for brand names or abbreviations)

Reply:
> The capitalization should be only the first letter capitalized (except for brand names or abbreviations)

Most entities appear to favor [Title Case](https://en.wikipedia.org/wiki/Title_case), at least from a quick scan through my entity list. In any case, if we treat the Home Assistant default entities as authoritative, it matches your style so lets go with that!

Reply:
Yea maybe because you haven't set them up already. Since 2023.8 were at 200-250/380 config flow integrations so if you set it up again you might see a difference :)